adage/analyze/management/commands/import_gene_network.py

```diff
@@ -153,31 +164,42 @@ def check_and_import(file_handle, ml_model):
         # If tokens[1] does not match one and only one gene's
         # systematic_name in the database, skip the line.
         try:
-            gene2 = find_gene(tokens[1])
+            gene2 = find_gene(tokens[1], cached_genes)
         except Exception as e:
             logger.warning("Input file line #%d will be skipped: %s",
                            line_index + 1, e)
             continue
 
         # Check whether the triplet (ml_model, gene1, gene2) is unique.
-        if not unique_together(ml_model, gene1, gene2):
+        if check_unique and not unique_together(ml_model, gene1, gene2):
             raise Exception("Input file line #%d: (%s, %s, %s) is not unique "
```

Comment:
No. If this exception is raised, it will terminate the program and roll back the database, because we would like the users to go back and check the input file immediately.

Reply:
Ok, nice!